big-int/src/operators/multiply.cpp

```diff
@@ -35,17 +35,38 @@ namespace libbig
 {
     largeInt largeInt::operator*(const largeInt &next_number) {
 
-        auto append_zeroes = [](const largeInt &x, const int factor) {
-            int k;
+        using Iterator = std::string::reverse_iterator; 
+
+        auto remove_zeroes = [] (const largeInt &x) {
             largeInt Answer = x;
-            for(k = 0; k < factor; k++) {
+            std::reverse(Answer.number.begin(), Answer.number.end());
+            for(Iterator j=Answer.number.rbegin(); j != Answer.number.rend() - 1; ++j) {
+                if((*j) == '0') {
+                    Answer.number.pop_back();
+                }
+                else {
+                    break;
+                }
+            }
+            std::reverse(Answer.number.begin(), Answer.number.end());
```

Comment:
This can be done in more optimized way.
just keep track of the previous digits until you find a digir greater than `0`
if digit found is greater than `0` then consider the string from this point till end and assign this to `Answer.number`